sdk/appconfiguration/azure-appconfiguration-provider/azure/appconfiguration/provider/aio/_azureappconfigurationproviderasync.py

```diff
@@ -226,11 +235,113 @@ class AzureAppConfigurationProvider(Mapping[str, Union[str, JSON]]):
+                            key,
+                            label,
+                        )
+                        await self._load_all(**kwargs)
+                        self._refresh_on[(key, label)] = updated_sentinel.etag
+                        self._refresh_timer.reset()
+                        return
+        except (ServiceRequestError, ServiceResponseError) as e:
+            logging.debug("Failed to refresh, retrying: %r", e)
+            self._refresh_timer.retry()
+        except HttpResponseError as e:
+            # If we get an error we should retry sooner than the next refresh interval
+            self._refresh_timer.retry()
+            if _is_retryable_error(e):
+                return
+            if self._on_refresh_error:
+                await self._on_refresh_error(e)
+                return
+            raise
+        except Exception as e:
+            if self._on_refresh_error:
+                await self._on_refresh_error(e)
+                return
+            raise
+
+    async def _load_all(self, **kwargs):
+        configuration_settings = {}
+        for select in self._selects:
+            configurations = self._client.list_configuration_settings(
+                key_filter=select.key_filter, label_filter=select.label_filter, **kwargs
+            )
+            async for config in configurations:
+                key = self._process_key_name(config)
+                value = self._process_key_value(config)
+
+                if isinstance(config, FeatureFlagConfigurationSetting):
+                    feature_management = configuration_settings.get(FEATURE_MANAGEMENT_KEY, {})
+                    feature_management[key] = value
+                    if FEATURE_MANAGEMENT_KEY not in configuration_settings:
+                        configuration_settings[FEATURE_MANAGEMENT_KEY] = feature_management
+                else:
+                    configuration_settings[key] = value
+                # Every time we run load_all, we should update the etag of our refresh sentinels
+                # so they stay up-to-date.
+                # Sentinel keys will have unprocessed key names, so we need to use the original key.
+                if (config.key, config.label) in self._refresh_on:
+                    self._refresh_on[(config.key, config.label)] = config.etag
+        self._dict = configuration_settings
+
+    def _process_key_name(self, config):
+        trimmed_key = config.key
+        # Trim the key if it starts with one of the prefixes provided
+        for trim in self._trim_prefixes:
+            if config.key.startswith(trim):
+                trimmed_key = config.key[len(trim) :]
+                break
+        if isinstance(config, FeatureFlagConfigurationSetting) and trimmed_key.startswith(FEATURE_FLAG_PREFIX):
+            return trimmed_key[len(FEATURE_FLAG_PREFIX) :]
+        return trimmed_key
+
+    def _process_key_value(self, config):
```

Comment:
This method needs to be async.

Reply:
And I assume we need await on this now?

Reply:
Yep :)
I'm actually surprised nothing showed up in testing, as this would have been returning a coroutine that never got awaited.

Reply:
We have an issue live tests with key vault, they never seem to work. That might have something to do with it.